common/priorityQueue.py

```diff
@@ -0,0 +1,84 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
+        self._size = 0
+        self._id = 0
+
+        # a unique sender is determined by the path to the sender's emails
+        self._uniqueSenders = set([])
+
+    def getLength(self):
+        return self._size
+
+    # returns the popped item
```

Comment:
Yes, that is true. I modified the code to stop returning a value and added comments on what push does.